iot/iot-tenant-cleaner/src/main/java/io/enmasse/iot/tools/cleanup/InfinispanTenantCleaner/InfinispanTenantCleaner.java

```diff
@@ -0,0 +1,181 @@
+            deviceconProvider.start();
+            devicesConnectionCache = deviceconProvider.getDeviceStateCache();
+        } catch (Exception e) {
+            log.error("Unable to access Infinispan caches.", e.getCause());
+            startPromise.fail(e);
+            return;
+        }
+
+        // Query and delete entries in devicesCache
+        QueryFactory queryFactory = Search.getQueryFactory(devicesCache);
+        Query query = queryFactory.from(DeviceInformation.class)
+                .select(Expression.property("deviceId"))
+                .having("tenantId")
+                .eq(tenantIdToClean)
+                .build();
+
+        List<String[]> matches = query.list();
+        matches.forEach(deviceInfo -> {
+            //fixme
+            DeviceKey key = DeviceKey
+                    .deviceKey(TenantHandle.of(tenantIdToClean, tenantIdToClean), deviceInfo[0]);
+            devicesCache.remove(key);
+
+            DeviceConnectionKey conKey = new DeviceConnectionKey(tenantIdToClean, deviceInfo[0]);
+            devicesConnectionCache.remove(conKey);
+        });
+    }
+
+    public void stop(Future stopFuture) {
+        // Stop the cache managers and release all resources
+        try {
+            deviceconProvider.stop();
+            mgmtProvider.stop();
+        } catch (Exception n) {
+            log.warn("Could not properly release cache manager resources in infinispan");
+        }
+    }
+
+    public Future<Void> configure() {
+
+        Future<Void> configured = Future.future();
+
+        ConfigRetriever retriever = ConfigRetriever.create(vertx, new ConfigRetrieverOptions()
+                .addStore(new ConfigStoreOptions()
+                        .setType("file")
+                        .setFormat("yaml")
+                        .setConfig(new JsonObject().put("path", pathToConfig)))
+                .addStore(new ConfigStoreOptions()
+                        .setType("env")));
+
+        retriever.getConfig(json -> {
+            if (json.failed()) {
+                log.error("Failed to read configuration.", json.cause());
+                configured.fail(json.cause());
+            } else {
+                JsonObject config = json.result();
+
+                tenantIdToClean = config.getString("iotProject");
+
+                JsonObject configInfinispan = config.getJsonObject("infinispan", new JsonObject());
+                System.out.println(configInfinispan.encodePrettily());
+                infinispanHost = configInfinispan.getString("host");
+                infinispanPort = configInfinispan.getInteger("port", 11222);
+                infinispanUsername = configInfinispan.getString("username");
+                infinispanPassword = configInfinispan.getString("password");
+                infinispanSaslServer = configInfinispan.getString("saslServerName");
+                infinispanSaslRealm = configInfinispan.getString("saslRealm");
+
+                try {
```

Comment:
I don't understand why you use the future and the try/catch to handle errors as the same time. I think using a plain exception may be ok, if the method is ensured to not block.

Reply:
this is a mistake, I should have removed the try/catch when switching to using the futures.